### ISSUE TITLE
Notifications: Allow to disable message preview

### ISSUE THREAD

**Issue by thomasst** [ChatSecure/ChatSecure-iOS#99]
My phone is password protected and I don't want anyone to be able to read my messages when I leave it unattended but I still want to be notified when I have new messages.

There could be three options on what to show in the notification:
- Only show me that I have new messages (or how many)
- Only show me who the sender is (iMessage lets you do this)
- Show both sender and message preview

I'd be most happy with the first option.

**Comment by mwlczk** [comment]
1st option should be the default for encrypted messages/chats. 
A previewed message is processed via iOS and thus other apps might be able to read the content!?

**Comment by chrisballinger** [comment]
At least on my iPhone X running iOS 11.3, you can disable notification previews when the phone is locked via iOS Settings -> Notifications.

> A previewed message is processed via iOS and thus other apps might be able to read the content!?

False

**Comment by mwlczk** [comment]
@chris: i mean something like this (OK it is signal and it is a Mac OS) https://motherboard.vice.com/amp/en_us/article/kzke7z/signal-disappearing-messages-are-stored-indefinitely-on-mac-hard-drives
This might be similar in iOS. Whenever iOS needs to show a notification it might store/process the content. So it would be better not to show the content to iOS for message preview of encrypted messages. :relaxed:

**Comment by chrisballinger** [comment]
Yes, but iOS apps are sandboxed and cannot read arbitrary system files like they can on macOS. Keep your iOS system version up to date and your risk is very minimal of a sandbox escape.

I'm not opposed to the idea if someone else submits a pull request, but time and resources are tight so this feature won't happen any time soon without some outside initiative.
